specs/AppLifecycle/Restart/restartApi.markdown

```
@@ -0,0 +1,259 @@
+<li><p>Kind: Gets the reason that this app is being activated. This is of type enum ActivationKind</p></li>
+</ul>
+<p>PreviousExecutionState: Gets the execution state of the app before this activation.</p></td>
+</tr>
+</tbody>
+</table>
+
+## New API Details
+
+### RequestRestartNow
+
+<table>
+<colgroup>
+<col style="width: 41%" />
+<col style="width: 53%" />
+<col style="width: 5%" />
+</colgroup>
+<thead>
+<tr class="header">
+<th>New API </th>
+<th>Description</th>
+</tr>
+</thead>
+<tbody>
+<tr class="odd">
+<td>static AppRestartFailureReason <strong>RequestRestartNow</strong>(string restartArguments)</td>
+<td><p>Equivalent to CoreApplication.RequestRestartAsync, except that it is <em>not</em> async.
+<ul>
+<li>If the restart fails, but the user subsequently launches the app manually, the app will launch normally and no restart arguments will be passed. </li>
+<li>If the app has any in-process background tasks running when it calls this API, those tasks will be cancelled in the normal way. Out-of-process background tasks will not be affected. </li>
+</ul>
+ The restarted instance will be activated with Windows.ApplicationModel.Activation.LaunchActivatedEventArgs.</p>
+ <ul>
+<li>When the app is restarted, LaunchActivatedEventArgs.PreviousExecutionState will have the value Terminated so that the app can distinguish between a resume and a restart. </li>
+</ul>
+<p>If the request to restart is successful, the application will be restarted immediately. The user does not have to do anything else/handle a return value in any way.
+
+If the request fails due to a nonfatal reason, a failure reason is returned for why the restart didn’t successfully complete. This method leverages an existing enum (AppRestartFailureReason) which has the following relevant return fields.
+<ul>
+<li> InvalidUser: Could not restart for the specified user.</li>
+<li> RestartPending: A restart is already in progress.</li>
+<li> Other: Unspecified failure.</li>
```

Having an `Other` prevents us from adding failure reasons in the future in a non-breaking way. Suppose we want to report an error like `BadAroma`. But old apps expect to receive `Other`, so how do we choose whether to return `Other` or `BadAroma`?

Are we confident that we'll never need to add a new failure reason in the future? Should we document that apps should treat any unrecognized result the same as `Other`?